components/BottomNavigation/tests/snapshot/MDCBottomNavigationBarControllerSnapshotTests.m

```diff
@@ -0,0 +1,230 @@
+        *fixedContentChildVC;
+
+@end
+
+@implementation MDCBottomNavigationBarControllerSnapshotTests
+
+- (void)setUp {
+  [super setUp];
+
+  // Uncomment below to recreate all the goldens (or add the following line to the specific
+  // test you wish to recreate the golden for).
+  //  self.recordMode = YES;
+
+  self.navBarController = [[MDCBottomNavigationBarController alloc] init];
+  self.navBarController.navigationBar.barTintColor =
+      [UIColor.redColor colorWithAlphaComponent:(CGFloat)0.5];
+  self.navBarController.view.backgroundColor = UIColor.whiteColor;
+  self.navBarController.view.bounds = CGRectMake(0, 0, 240, 360);
+
+  self.fixedContentChildVC =
+      [[MDCBottomNavigationBarControllerSnapshotTestFixedContentChildViewController alloc] init];
+  self.fixedContentChildVC.view.backgroundColor = UIColor.blueColor;
+  self.fixedContentChildVC.tabBarItem.title = @"Fixed";
+
+  UICollectionViewFlowLayout *flowLayout = [[UICollectionViewFlowLayout alloc] init];
+  flowLayout.estimatedItemSize = kSizeOfCellInScrollableChild;
+  self.scrollableChildVC =
+      [[MDCBottomNavigationBarControllerSnapshotTestScrollableChildViewController alloc]
+          initWithCollectionViewLayout:flowLayout];
+  self.scrollableChildVC.collectionView.backgroundColor = UIColor.blueColor;
+  self.scrollableChildVC.tabBarItem.title = @"Scrollable";
+
+  self.navBarController.viewControllers = @[ self.scrollableChildVC, self.fixedContentChildVC ];
+  self.navBarController.navigationBar.unselectedItemTintColor = UIColor.blackColor;
+  self.navBarController.navigationBar.titleVisibility = MDCBottomNavigationBarTitleVisibilityAlways;
+}
+
+- (void)tearDown {
+  self.scrollableChildVC = nil;
+  self.fixedContentChildVC = nil;
+  self.navBarController = nil;
+
+  [super tearDown];
+}
+
+- (void)generateAndVerifySnapshot {
+  UIView *backgroundView = [self.navBarController.view mdc_addToBackgroundView];
+  [self snapshotVerifyView:backgroundView];
+}
+
+#pragma mark - Tests
+
+- (void)testNonScrollingChildViewControllerDefault {
+  // When
+  self.navBarController.selectedViewController = self.fixedContentChildVC;
+  [self.navBarController.view layoutIfNeeded];
+
+  // Then
+  [self generateAndVerifySnapshot];
+}
+
+- (void)testScrollableChildViewControllerDefault {
+  // When
+  self.navBarController.selectedViewController = self.scrollableChildVC;
+  [self.navBarController.view layoutIfNeeded];
+
+  // Then
+  [self generateAndVerifySnapshot];
+}
+
+- (void)testScrollableChildViewControllerScrolledToBottom {
+  // Given
+  self.navBarController.selectedViewController = self.scrollableChildVC;
+  // Forces the layout system to update the layout margins and correctly position the
+  // scroll view within the content view of the Bottom Navigation bar.
+  UIView *backgroundView = [self.navBarController.view mdc_addToBackgroundView];
+  [backgroundView drawViewHierarchyInRect:backgroundView.bounds afterScreenUpdates:YES];
```

Comment:
@bryanoltman @yarneo Any ideas why *this* seems to be required to get the `layoutMargins` to be applied from the superview?  It didn't work when I only had `layoutIfNeeded`.

Reply:
Maybe we need to set `usesDrawViewHierarchyInRect` to `YES` on `FBSnapshotTestController`

Reply:
It might be worth doing in general, but in this case the oddness appears to be that UICollectionView is responding to its superview's `layoutMargins` or something else. I tried setting them to `{0, 0, 0, 0}` but that didn't have any effect.  I'm not thrilled that I have to render the view in the middle of the test (and then render again), so I'm hoping to discover what I can do more directly to achieve the same effect.  

I'm glad that it works, but  I'm not happy that I don't know *why* it works.